onecodex/models/collection.py

```diff
@@ -347,23 +353,31 @@ def _collate_metadata(self):
                 classification_id = None
             sample = obj.sample if isinstance(obj, Classifications) else obj
 
+            # Standard/structured metadata fields get priority over custom metadata fields in the
+            # event of a name clash. If the user is allowed to override certain metadata fields like
+            # `sample_id`, `metadata_id`, and `classification_id`, downstream code will break in
+            # strange ways.
             m = sample.metadata
+            metadatum = {}
+            metadatum.update(m.custom)
 
             if DEFAULT_FIELDS is None:
                 DEFAULT_FIELDS = list(m.__class__.model_fields.keys())
                 DEFAULT_FIELDS.remove("field_uri")
                 DEFAULT_FIELDS.remove("sample")
                 DEFAULT_FIELDS.remove("custom")
 
-            metadatum = {f: getattr(m, f) for f in DEFAULT_FIELDS}
+            for f in DEFAULT_FIELDS:
+                metadatum[f] = getattr(m, f)
+
             metadatum["classification_id"] = classification_id
             metadatum["sample_id"] = sample.id
             metadatum["metadata_id"] = m.id
             metadatum["created_at"] = sample.created_at
             metadatum["filename"] = sample.filename
             metadatum["project"] = getattr(sample.project, "name", "")
+            metadatum["sample_name"] = m.name or sample.filename or "N/A"
```

Comment:
Should this match `Sample.display_name` (in the DB model)? Maybe we should just expose that via v1 API?